--- dsdt.dsl.org	2016-02-10 22:17:22.887989996 -0800
+++ dsdt.dsl	2016-02-12 15:41:54.736377560 -0800
@@ -25156,10 +25156,14 @@
             Method (_Q0A, 0, NotSerialized)  // _Qxx: EC Query
             {
                 P80H = 0x0A
-                If (^^^^WMI.HKDR)
+
+                If (^^^^WMI.GCMD (0x00, 0x09, 0x00))
                 {
-                    ^^^^WMI.EVNT = OEM4 /* \_SB_.PCI0.LPCB.EC__.OEM4 */
-                    Notify (WMI, 0xD0) // Hardware-Specific
+                    ^^^^WMI.SCMD (0x00, 0x2A, 0x00)
+                }
+                else
+                {
+                    ^^^^WMI.SCMD (0x00, 0x2A, 0x01)
                 }
             }
 
